torch/_inductor/cudagraph_trees.py

```diff
@@ -592,7 +604,10 @@ def __init__(
             self.recorded_liveness_before_graph = curr_liveness
             self.expected_dead_indices_before_graph = different_indices
 
-        recording_inputs = self._allocate_recording_inputs(inputs)
+        recording_inputs = self._allocate_and_copy_recording_inputs(inputs)
+        # recording inputs will copy over memory, so we can free non recording inputs
+        inputs.clear()
+        del inputs
```

Comment:
this del doesn't actually do anything right

Reply:
Correct - i just added it to make sure we no longer use it in this scope